apps/src/templates/teacherDashboard/AssignmentSelector.jsx

```diff
@@ -0,0 +1,87 @@
+import React, { Component, PropTypes } from 'react';
+import _ from 'lodash';
+import { assignmentShape } from './shapes';
+
+export default class AssignmentSelector extends Component {
+  static propTypes = {
+    courseId: PropTypes.number,
+    scriptId: PropTypes.number,
+    validCourses: PropTypes.arrayOf(assignmentShape).isRequired,
+    validScripts: PropTypes.arrayOf(assignmentShape).isRequired,
+  };
+
+  getSelectedAssignment() {
+    const assignment = this.assignments[this.root.value];
+    return {
+      course_id: assignment.course_id,
+      script_id: assignment.script_id
+    };
+  }
+
+  render() {
+    const { courseId, scriptId, validCourses, validScripts } = this.props;
+
+    // TODO(bjvanminnen): It's not clear that all of this data manipulation belongs
+    // here. If we move our data to redux, it likely belongs there. Otherwise, it
+    // may still make more sense for this to all happen before instantiating our
+    // React tree.
```

Comment:
Agreed.  This all feels like it should move to `componentWillReceiveProps` or something if it stays in React.  Either way, I'd extract a bunch of the manipulation here to helper methods - easy to move around, easy to test.